pkgs/development/libraries/optparse-bash/default.nix

```diff
@@ -0,0 +1,63 @@
+{ stdenvNoCC
+, lib
+, fetchFromGitHub
+, bash
+, gnused
+, gawk
+, coreutils
+}:
+
+stdenvNoCC.mkDerivation {
+  pname = "optparse-bash-unstable";
+  version = "2017-06-11";
+
+  src = fetchFromGitHub {
+    owner = "nk412";
+    repo = "optparse";
+    rev = "0dfc5a3344feafc818b1fb051e4bebd195d18893";
+    sha256 = "sha256-fFDKU4ZQxrDokxwjHpXQaL3XGwdrSPugQSJbXMi6tJY=";
+  };
+
+  postPatch = ''
+    substituteInPlace optparse.bash \
+      --replace sed "${gnused}/bin/sed" \
+      --replace awk "${gawk}/bin/awk" \
+      --replace printf "${coreutils}/bin/printf"
+'';
+
+  dontBuild = true;
+
+  doCheck = true;
+
+  checkInputs = [ bash ];
+
+  # `#!/usr/bin/env` isn't okay for OfBorg
+  # Need external bash to run
+  checkPhase = ''
+    runHook preCheck
+    bash ./sample_head.sh -v --file README.md
+    runHook postCheck
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    mv optparse.bash $out/bin/
```

Comment:
./results/optparse-bash/bin/optparse.bash 
bash: ./results/optparse-bash/bin/optparse.bash: Permission denied

Use '[install](https://www.geeksforgeeks.org/install-command-in-linux-with-examples/)' command to set permission.

Reply:
@superherointj 

`optparse.bash` is meant to be sourced (as a "header file" of BASH functions) instead of being executed.

I put it inside `$out/bin` so that users can source it like

```BASH
source optparse.bash
```

instead of

```
source /nix/store/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-optparse-bash-0000-00-00/libexec/optparse.bash
```

Is there a cleaner approach of doing so?

Reply:
If you use it in a nix package then you can substitute the store path with ``${optparse}/libexec/optparse.bash``.